pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -0,0 +1,854 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import logging
+import re
+import sys
+import pyomo.common
+from pyutilib.misc import Bunch
+from pyutilib.services import TempfileManager
+from pyomo.core.expr.numvalue import is_fixed
+from pyomo.core.expr.numvalue import value
+from pyomo.repn import generate_standard_repn
+from pyomo.solvers.plugins.solvers.direct_solver import DirectSolver
+from pyomo.solvers.plugins.solvers.direct_or_persistent_solver import DirectOrPersistentSolver
+from pyomo.core.kernel.objective import minimize, maximize
+from pyomo.core.kernel.component_set import ComponentSet
+from pyomo.core.kernel.component_map import ComponentMap
+from pyomo.opt.results.results_ import SolverResults
+from pyomo.opt.results.solution import Solution, SolutionStatus
+from pyomo.opt.results.solver import TerminationCondition, SolverStatus
+from pyomo.opt.base import SolverFactory
+from pyomo.core.base.suffix import Suffix
+import pyomo.core.base.var
+
+
+logger = logging.getLogger('pyomo.solvers')
+
+
+class DegreeError(ValueError):
+    pass
+
+
+def _is_numeric(x):
+    try:
+        float(x)
+    except ValueError:
+        return False
+    return True
+
+
+@SolverFactory.register('mosek', doc='Direct python interface to Mosek')
+class MosekDirect(DirectSolver):
```

Comment:
I believe the interface is missing the `.available()` method.  That would be the standard approach for detecting if the solver interface was available and properly licensed.

Reply:
The base class implements the `available()` method based on the instance variable `self._python_api_exists`, which the mosek plugin sets.

I can confirm that `pyomo help --solvers` works with and without mosek being available. Is there another command you want me to test?

Reply:
The problem I am seeing is:
```
>>> from pyomo.environ import *
>>> s = SolverFactory('mosek')
>>> s.available()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[...]/pyomo/solvers/plugins/solvers/direct_or_persistent_solver.py", line 299, in available
    "plugin").format(type(self)))
pyutilib.common._exceptions.ApplicationError: No Python bindings available for <class 'pyomo.solvers.plugins.solvers.mosek_direct.MosekDirect'> solver plugin
```
I would expect `available()` to return False.

Reply:
I agree it should return False, but that has been the default behavior of `available()` since it was implemented. I believe you can pass `exception_flag=False` to get the more rational behavior, which we should make the default in a new PR.